File: source/com/barrybecker4/optimization/viewer/NavigationListener.java
// Copyright by Barry G. Becker, 2012. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.optimization.viewer;

import javax.vecmath.Point2d;

/**
 * Implemented by something that listens for navigation events.
 * @author Barry Becker
 */
public interface NavigationListener {

    enum Direction {UP, DOWN, LEFT, RIGHT}

    void pan(Point2d offset);
    void zoomIn();
    void zoomOut();
}

File: scala-source/com/barrybecker4/optimization/viewer/ui/NavigationListener.scala
// Copyright by Barry G. Becker, 2000-2018. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.optimization.viewer.ui

import javax.vecmath.Point2d


/**
  * Implemented by something that listens for navigation events.
  * @author Barry Becker
  */
trait NavigationListener {

  def pan(offset: Point2d): Unit

  def zoomIn(): Unit

  def zoomOut(): Unit
}
